data_steward/validation/main.py

```diff
@@ -174,6 +176,8 @@ def _upload_achilles_files(hpo_id=None, folder_prefix='', target_bucket=None):
             raise RuntimeError(
                 f"Either hpo_id or target_bucket must be specified")
         target_bucket = storage_client.get_hpo_bucket(hpo_id)
+    else:
+        target_bucket = storage_client.bucket(target_bucket.name)
```

Comment:
If we use `target_bucket.name` here, we assume `target_bucket` to be a bucket object. However, we are passing `str` to `target_bucket` when we call this function. So, we need to keep `storage_client.bucket(target_bucket)`. Otherwise, the problem described in `DC-2446` will remain. 

The test assumes the wrong data type, which is causing this error.

https://github.com/all-of-us/curation/blob/develop/tests/integration_tests/data_steward/validation/main_test.py#L261

The test needs `.name` added to represent the real-life use case. 
```
        main._upload_achilles_files(hpo_id=None,
                                    folder_prefix=folder_prefix,
                                    target_bucket=bucket_nyc.name)
```

Reply:
Thanks, @hiro-mishima !